core/src/main/scala/kafka/server/ClientQuotaManager.scala

```diff
@@ -139,11 +139,14 @@ class ClientQuotaManager(private val config: ClientQuotaManagerConfig,
    * we need to add a delay of X to W such that O * W / (W + X) = T.
    * Solving for X, we get X = (O - T)/T * W.
    */
-  private def throttleTime(metricValue: Double, config: MetricConfig): Int = {
+  private def throttleTime(clientMetric: KafkaMetric, config: MetricConfig): Int = {
+    // Casting to Rate because we only use Rate in Quota computation
+    val rateMetric = clientMetric.measurable().asInstanceOf[Rate]
```

Comment:
These metrics are created as Rate within the same ClientQuotaManager class.. so you could say it is being enforced. Since the logic remains private to this implementation, I tend to think measurableAsRate is not necessary. 

Thoughts? I'll address this if you feel strongly about it.